instagram-app/src/App.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+//import logo from './logo.svg';
+import './App.css';
+import dummyData from './dummy-data';
+import BarSearch from './components/SearchBar/searchbar';
+import PostCont from './components/PostContainer/postcontainer';
+import '../src/components/login.css';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+      this.state = {
+        mydata: dummyData,
+        local_user: "",
+        post_comment: "",
+      }
+  }
+
+  //function to save the state of the likes
+
+  triggerlikes = (e, username) => {
+    e.preventDefault();
+      this.setState(preState => {
+        const updatedlike = preState.mydata.map(mydataupdated => {
+          if (mydataupdated.username === username) {
+            mydataupdated.liked = !mydataupdated.liked
+            mydataupdated.liked ? mydataupdated.likes = mydataupdated.likes + 1 : mydataupdated.likes = mydataupdated.likes - 1
+          }
+          return mydataupdated;
+        })
+        return {
+          mydata : updatedlike
+        }
+      })
+  }
+
+  //function to trigger user form if there is no local username state
+  postaction = (e, username) => {
```

Comment:
I would make this a higher order component and save the username to localStorage that way there is a little bit of persistence on reload so the user doesn't have to login on every reload. check for username on localStorage, set it on state. If no username on state, trigger a login component that will ask a user to log in.